caddy.go

```diff
@@ -875,6 +875,16 @@ func ParseDuration(s string) (time.Duration, error) {
 	return time.ParseDuration(s)
 }
 
+// sliceContains returns true if needle is in haystack.
+func SliceContains(haystack []string, needle string) bool {
+	for _, s := range haystack {
+		if s == needle {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
There's no need this function be exposed outside Caddy. Let's move it to `internal`. Better now than later.

Reply:
Or we could use this new generic function: https://pkg.go.dev/slices#Contains